Upgrade Magnus to 0.6

matsadler opened the pull request:
Hey, I've [released a new version of Magnus](https://github.com/matsadler/magnus/releases/tag/0.6.0), and while I was preparing that I once again tested against wasmtime-rb.

These are the changes I made to get things working while I was doing that. I don't know if I'll have time to follow up on this PR, so please feel free to rework them yourself, cherry-pick commits, or just use this as inspiration.

jbourassa commented:
Thanks Mat, very helpful as always.

I'm currently travelling without a computer, but will pick this up once I'm back ~mid August, unless someone beats me to it.

matsadler commented:
@jbourassa I found some time over the holidays to get this rebased on main and CI passing. I think it's good to go now.

jbourassa reviewed (commented):
This looks great, thanks.

I punted on the initial review given I haven't been super active on this project and someone else picked it up in a separate PR, but that work looks stalled. I'm happy to merge this.

I put a few minor comments. I'm happy to make the changes if you'd like, but that implies opening new PRs with your work. Let me know what you prefer.

jbourassa commented:
I also ran the `func_call` benchmark and noticed a small speedup. No idea where it comes from, but we'll take it!

matsadler commented on ext/src/ruby_api/func.rs:
```diff
@@ -233,27 +222,28 @@ pub fn make_func_closure(
+        let callable = ruby.get_inner(callable);
 
-        match (callable.call(unsafe { rparams.as_slice() }), results.len()) {
+        match (callable.call(rparams), results.len()) {
```
Done, see #274.

matsadler commented on ext/src/ruby_api/store.rs:
```diff
@@ -296,10 +289,7 @@ impl<'a> StoreContextValue<'a> {
-                    Ok(e) => e,
-                    Err(e) => error!("{}", e),
-                })
+                .unwrap_or_else(|e| error!("{}", e))
```
It's been a while since I did it, but I think that was the conclusion I came to. Good to have it confirmed.

matsadler commented on ext/src/ruby_api/store.rs:
```diff
@@ -47,43 +51,25 @@ impl StoreData {
-                }
-                Error::Jump(_) => {}
+            if let Some(val) = error.value() {
+                marker.mark(val);
```
Yep.

matsadler commented:
I think I've addressed all the comments. Let me know if there was anything I missed.

> I also ran the `func_call` benchmark and noticed a small speedup. No idea where it comes from, but we'll take it!

In developing the 0.6 release of Magnus I did some profiling and added some `#[inline]`s to things I either missed, or mistakenly thought would be inlined without the annotation, so it could be that.

jbourassa reviewed (approved):
Thanks again
